test/WireMock.Net.Tests/StatefulBehaviorTests.cs

```diff
@@ -156,19 +157,20 @@ public class StatefulBehaviorTests : IDisposable
                 .RespondWith(Response.Create()
                     .WithBody("Test state msg 2"));
 
+            Thread.Sleep(500);
+
             // when / then
             string url = "http://localhost:" + _server.Ports[0];
-            var http = new HttpClient();
-            var responseNoState1 = await http.GetStringAsync(url + "/state1");
+            var responseNoState1 = new HttpClient().GetStringAsync(url + "/state1").Result;
             Check.That(responseNoState1).Equals("No state msg 1");
 
-            var responseNoState2 = await http.GetStringAsync(url + "/state2");
+            var responseNoState2 = new HttpClient().GetStringAsync(url + "/state2").Result;
             Check.That(responseNoState2).Equals("No state msg 2");
 
-            var responseWithState1 = await http.GetStringAsync(url + "/foo");
+            var responseWithState1 = new HttpClient().GetStringAsync(url + "/foo").Result;
             Check.That(responseWithState1).Equals("Test state msg 1");
 
-            var responseWithState2 = await http.GetStringAsync(url + "/foo");
+            var responseWithState2 = new HttpClient().GetStringAsync(url + "/foo").Result;
             Check.That(responseWithState2).Equals("Test state msg 2");

```

Comment:
Actually *sometimes*, I had some failing tests. So I decided to use a new HttpClient each time, which should works fine because it does not matter who calls the server; the server should track state.

The issue could be related to the order and if the mappings are correctly added before the clients connect maybe????